packages/core/src/transports/base.ts

```diff
@@ -0,0 +1,147 @@
+import { Envelope, EventStatus } from '@sentry/types';
+import {
+  disabledUntil,
+  eventStatusFromHttpCode,
+  isRateLimited,
+  makePromiseBuffer,
+  PromiseBuffer,
+  RateLimits,
+  rejectedSyncPromise,
+  resolvedSyncPromise,
+  serializeEnvelope,
+  updateRateLimits,
+} from '@sentry/utils';
+
+export const ERROR_TRANSPORT_CATEGORY = 'error';
+
+export const TRANSACTION_TRANSPORT_CATEGORY = 'transaction';
+
+export const ATTACHMENT_TRANSPORT_CATEGORY = 'attachment';
+
+export const SESSION_TRANSPORT_CATEGORY = 'session';
+
+type TransportCategory =
+  | typeof ERROR_TRANSPORT_CATEGORY
+  | typeof TRANSACTION_TRANSPORT_CATEGORY
+  | typeof ATTACHMENT_TRANSPORT_CATEGORY
+  | typeof SESSION_TRANSPORT_CATEGORY;
+
+export type TransportRequest = {
+  body: string;
+  category: TransportCategory;
+};
+
+export type TransportMakeRequestResponse = {
+  body?: string;
+  headers?: Record<string, string | null>;
```

Comment:
debating if I need to make the rate limiting headers mandatory here.

Reply:
Yeah I will change this to be non-optional

We allow `null` because most getters for headers return null if not defined. For example, https://developer.mozilla.org/en-US/docs/Web/API/Headers/get